src/utils/lesson-metadata.ts

```diff
@@ -64,3 +66,20 @@ const collectionIsPresent = (collection: {lessons: any[] | undefined}) => {
   // collection is considered present.
   return some(lessons) && some(collectionMetadata)
 }
+
+export const deriveDataFromBaseValues = ({path}: {path?: string}) => {
+  if (!isEmpty(path)) {
+    invariant(
+      path?.startsWith('/'),
+      'Path value must begin with a forward slash (`/`).',
+    )
```

Comment:
Move this function to the `lesson-metadata.ts` file.

1. Better able to test it now that it is here and exported.
2. Able to identify a typing improvement, from `{path: string}` to `{path?: string}` since `path` may not get passed in which is a key part of the conditional logic in here.